feat: Add Revenue analytics filters to Max AI

rafaeelaudibert opened the pull request:
We can now customize dates, filters and breakdowns in Revenue Analytics. This was implemented by copying what we already had for Session Replay and expanding the taxonomy agent slightly so that it can learn how to find the relevant revenue analytics properties.

Local tests seem to show this works well, but I'd appreciate some help figuring out whether I should write tests for this or not

<img width="838" height="105" alt="image" src="https://github.com/user-attachments/assets/c514e417-781b-44cd-811e-734024906e95" />

rafaeelaudibert committed:
feat: Add Revenue analytics filters to Max AI

We can now customize dates, filters and breakdowns in Revenue Analytics. This was implemented by copying what we already had for Session Replay and expanding the taxonomy agent slightly so that it can learn how to find the relevant revenue analytics properties.

Local tests seem to show this works well, but I'd appreciate some help figuring out whether I should write tests for this or not

rafaeelaudibert committed:
fix: Fix copy/paste

rafaeelaudibert committed:
fix: Fix types

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

denakorita commented on products/revenue_analytics/backend/max_tools.py:
```diff
@@ -0,0 +1,160 @@
+        - "revenue analytics" synonyms: "revenue", "MRR", "growth", "churn", and similar
+      * When the user asks to search for revenue analytics or revenue
+        - "search for" synonyms: "find", "look up", and similar
+    - When NOT to use the tool:
```
Did you add this because you saw it trigger insight creation?

denakorita commented on products/revenue_analytics/backend/max_tools.py:
```diff
@@ -0,0 +1,160 @@
+    def __init__(self, team: Team):
+        super().__init__(team)
+
+    def _get_custom_tools(self) -> list:
```
And if you do not want all the other taxonomy tools you can simply override the `_get_default_tools`

denakorita reviewed (changes requested):
Great work, I added a few comments for minor things to change

rafaeelaudibert commented on products/revenue_analytics/backend/max_tools.py:
I can, thanks :). I don't know how to test them, though, any tips?

rafaeelaudibert committed:
refactor: Use custom tool rather than tweaking native taxonomy

Rather than adding a new feature to the base taxonomy agent, let's simply expose fetching the revenue analytics values as a tool, this is much simpler

rafaeelaudibert committed:
fix: Fix flakey revenue analytics tests

Let's make sure we know it's 2024 to avoid flakies once we go to the next year

rafaeelaudibert commented on ee/hogai/eval/ci/max_tools/eval_session_replay_filter_generation.py:
@denakorita these are my changes. I couldn't run the evals myself because I started hitting rate limits on my OpenAI account, but they worked for me on revenue analytics

denakorita replied:
Hey, what is the status here?

rafaeelaudibert commented on ee/hogai/graph/taxonomy/toolkit.py:
@denakorita I extracted a lot of these functions to standalone ones to be able to use them in the Revenue Analytics prompts. Since the place where we add the prompts is a `TaxonomyAgentNode`, we don't have access to the internal format functions. By exposing them here I can easily use them everywhere else.

rafaeelaudibert replied:
Another update, I've moved them to a separate `format.py` file

rafaeelaudibert committed:
refactor: Extract function to new `format` file

Also, avoid formatting directly on the `prompts` file, do it on the tools instead

skoob13 reviewed (approved):
LGTM

rafaeelaudibert committed:
fix: Remove type

This type is pretty bad because it's generic but it includes a lot of keys we don't care about. Just removing it makes the code typecheck properly

rafaeelaudibert commented on ee/hogai/graph/taxonomy/prompts.py:
```diff
@@ -34,6 +34,7 @@
     Use `retrieve_entity_properties` with entity="[group_name]" to get the list of all available group properties.
 </group>
 </entity>
+
```
I added it because there were whitespaces between the other XML tags, but of course :)

denakorita replied:
Ah got it

rafaeelaudibert committed:
refactor: Revert new whitespace

denakorita reviewed (approved):
Great, let's ship it!
